app/controllers/application_controller.rb

```diff
@@ -1,5 +1,18 @@
 class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
-  protect_from_forgery with: :exception
+  #protect_from_forgery with: :exception
+
+   def current_user
+    if session[:user_id]
+      User.find(session[:user_id])
+    else
+      return nil
```

Comment:
Return nil is superfluous. This method could be
User.find(session[:user_id]) if session[:user_id]
as a one liner.

BUT there's another problem. If there's a user id in the session that doesn't match a user in the database, User.find will throw an error. 
User.find_by(id: session[:user_id]) if session[:user_id]
would be better

If